### ISSUE TITLE
Organic maps gives a bit to much weight to cycleways

### ISSUE THREAD

**Issue by vincentvd1** [organicmaps/organicmaps#1772]
I have noticed that organic maps gives a bit to much weight to cycleways during navigation for bicycles. See the example below:
![Organic maps navigation](https://i.imgur.com/d6Z9lHj.png)
![Graphhopper naviation](https://i.imgur.com/uqncjDs.png)

[Graphhopper location](https://graphhopper.com/maps/?point=52.253405%2C6.182288&point=52.247599%2C6.197973&locale=nl-nl&elevation=true&profile=bike&use_miles=false&selected_detail=Elevation&layer=Omniscale)

The route by organic maps results in an additional 500m where logically, I would use the route by graphhopper this time. It is more direct where in my opinion the shorter route outweights the additional 500m that is over cycleways.

I have to note though that this preference for cycleways in general results in better routes then graphhopper. In the example below, organic maps routes me over cycleways near major routes where graphbopper gives me a route through the residential area. Using the cycleways gives a much nicer bicycle experience and almost no additional time.
![correct route](https://i.imgur.com/RTDPNbo.png)
![Graphhopper route](https://i.imgur.com/HQzsaAy.png)

**Comment by biodranik** [comment]
Do you think that providing alternative routes will solve this issue? Or simply slightly reducing the weight should be enough?

**Comment by vincentvd1** [comment]
@biodranik 	With alternative routes, do you mean like Google maps showing multiple routes where you can select the one you like? I peronally don't like that. It clutters the navigation. I would like to trust on the navigation that it gives me the most optimal route. Otherwise you would have to assess the alternatives during or prior to navigation.

Reducing the weight might be an option that can be looked at. If it is possible to reduce the weight while still providing the same routes in other cases (like the second example), that would be fine to me. But that is something that needs to be tested.

**Comment by biodranik** [comment]
When the app shows you only one route, it's hard to decide how good is it. If it shows "the shortest", "the fastest", and "by cycleway", then at least you can choose the best one for you.

**Comment by vincentvd1** [comment]
I don't agree entirely. When I enter a destination in our car navigation, it gives 1 route. Imagine if you would first have to look at each route which you like the most. For shorter routes this is doable but for longer routes not. The most optimal route should be a balanced one. By reports like this one, you can fine tune the algorithm. 

It is also possible to make a setting whether you want to see alternative routes.

**Comment by biodranik** [comment]
The problem roots are in the imperfect OSM data. Fine-tuning the algorithm won't help with it. From the user perspective, there is no overhead: you still have some ("the best") route selected by default, while seeing several alternate ones with an option to touch and activate them (or ignore and just press "GO!").

**Comment by vincentvd1** [comment]
Thanks for the clearification. About fine tuning, you as developer know the algorithm the best so if fine tuning (lowering the weight) does not help in your opinion, okay. Then the best alternative is indeed showing multiple routes which you can select (and  a balanced route selected as default).

**Comment by chereskata** [comment]
> The problem roots are in the imperfect OSM data. Fine-tuning the algorithm won't help with it. From the user perspective, there is no overhead: you still have some ("the best") route selected by default, while seeing several alternate ones with an option to touch and activate them (or ignore and just press "GO!").

This is THE feature i am missing atm :)

**Comment by matkoniecz** [comment]
> The problem roots are in the imperfect OSM data.

what about https://private-user-images.githubusercontent.com/899988/323778942-fae51b62-afcb-41ae-9d58-e1b5384bd83a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAzMDQ3NjAsIm5iZiI6MTczMDMwNDQ2MCwicGF0aCI6Ii84OTk5ODgvMzIzNzc4OTQyLWZhZTUxYjYyLWFmY2ItNDFhZS05ZDU4LWUxYjUzODRiZDgzYS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDMwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAzMFQxNjA3NDBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01NjI2YzY0MGQ4MTQ5ZTAyZWIzMjY4ZjVjMTU3OWI3NzFlYTkwMmZiZWJlMjk3MTU1NmYwZDg3MzY2ZTQ0Mjg2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.gxGaukAvLtbz3bgpZkzU9Nnrn8TdIA4sQPvk57nbyyA case?

here https://github.com/organicmaps/organicmaps/issues/1772 is part of the problem, but even without crossing this route would be still silly.

**Comment by matkoniecz** [comment]
> When the app shows you only one route, it's hard to decide how good is it. If it shows "the shortest", "the fastest", and "by cycleway", then at least you can choose the best one for you.

Though maybe having separate "fast bicycle" and "safe bicycle" would make sense. (bicycle with shield for icon? child bicycle? Bicycle with speed/wind dashes behind it for fast one?)

**Comment by pastk** [comment]
> what about

could you please re-upload the pic (it doesn't work for me) and/or provide coords?

**Comment by pastk** [comment]
> I have to note though that this preference for cycleways in general results in better routes then graphhopper. In the example below, organic maps routes me over cycleways near major routes where graphbopper gives me a route through the residential area. Using the cycleways gives a much nicer bicycle experience and almost no additional time.

These two routes differ by ~50m only (check GraphHopper vs Valhalla https://www.openstreetmap.org/directions?engine=graphhopper_bicycle&route=51.18346%2C6.00122%3B51.19257%2C5.99385#map=16/51.18800/5.99714), so its very safe OM would still prefer cycleways even if the preference for them would be reduced significantly.

Do you perhaps have some other sample cases where there is a bigger difference between a preferred cycleways route vs a noticeably shorter but less pleasant road route?

**Comment by matkoniecz** [comment]
> Do you perhaps have some other sample cases where there is a bigger difference between a preferred cycleways route vs a noticeably shorter but less pleasant road route?

should I try find now before https://github.com/organicmaps/organicmaps/pull/9692 is merged or would it be more useful to wait for it to become available for me and then report some silly routes?

**Comment by pastk** [comment]
@matkoniecz yeap it'd be good to have a robust "cycleway detour is preferred" case
(to replace the RussiaMoscowSevTushinoParkPreferingBicycleWay case)
